conjureplugin/config/config.go

```diff
@@ -44,54 +46,75 @@ func ToConjurePluginConfig(in *ConjurePluginConfig) *v1.ConjurePluginConfig {
 // Currently, if multiple Conjure projects have the same output directory (after normalization using filepath.Clean),
 // this is considered to be warning. The returned warning is an error created using errors.Join that contains one error
 // per output path shared by multiple projects.
-func (c *ConjurePluginConfig) ToParams() (_ conjureplugin.ConjureProjectParams, warnings []error, err error) {
+func (c *ConjurePluginConfig) ToParams() (_ conjureplugin.ConjureProjectParams, warnings []error, _ error) {
 	sortedKeys := slices.Sorted(maps.Keys(c.ProjectConfigs))
 
-	seenDirs := make(map[string][]string)
+	conflicts := ToConjurePluginConfig(c).OutputDirConflicts()
+
 	params := make(map[string]conjureplugin.ConjureProjectParam)
+	var conflictErrs error
 	for _, key := range sortedKeys {
+		if err := validate.ValidateProjectName(key); err != nil {
+			return conjureplugin.ConjureProjectParams{}, nil, err
+		}
+
 		currConfig := c.ProjectConfigs[key]
 
-		// normalize outputDir
-		outputDir := filepath.Clean(currConfig.OutputDir)
-		seenDirs[outputDir] = append(seenDirs[outputDir], key)
+		outputDir := currConfig.ResolvedOutputDir(key)
+
+		if !currConfig.SkipDeleteGeneratedFiles && len(conflicts[key]) > 0 {
+			conflictErrs = errors.Join(conflictErrs, errors.Join(conflicts[key]...))
```

Comment:
I think that nested joins will actually double-wrap (I don't think they unwrap before joining), so I think it would be better to accumulate all the errors into a slice and then errors.Join on the slice once